app/assets/javascripts/discourse/app/components/desktop-notification-config.js

```diff
@@ -83,9 +83,13 @@ export default Component.extend({
   isEnabled: or("isEnabledDesktop", "isEnabledPush"),
 
   isPushNotificationsPreferred() {
-    if (!this.site.mobileView) {
+    if (
+      !this.site.mobileView &&
+      !this.siteSettings.enable_desktop_push_notifications
```

Comment:
```suggestion
      this.site.mobileView ||
      this.siteSettings.enable_desktop_push_notifications
```

Would this be more readable?